qiskit/backends/_qeremote.py

```diff
@@ -0,0 +1,112 @@
+import time
+import logging
+from qiskit.backends._basebackend import BaseBackend
+from qiskit.backends._backendutils import get_backend_configuration
+from qiskit import _openquantumcompiler as openquantumcompiler
+from qiskit._result import Result
+from IBMQuantumExperience.IBMQuantumExperience import IBMQuantumExperience
+
+logger = logging.getLogger(__name__)
+
+class QeRemote(BaseBackend):
+    def __init__(self, configuration=None):
+        """Initialize remote backend for IBM Quantum Experience.
+
+        Args:
+
+        """
+        if configuration is None:
+            self._configuration = get_backend_configuration(backend_name)
+        else:
+            self._configuration = configuration
+        self._configuration['local'] = False
+
+    def run(self, q_job):
+        """Run jobs
+
+        Args:
+            q_job (QuantumJob): job to run
+
+        Returns:
+            Result object.
+
+        Raises:
+            ResultError: if the api put 'error' in its output
+        """
+        self._qobj = q_job.qobj
+        wait = q_job.wait
+        timeout = q_job.timeout
+        api_jobs = []
+        for circuit in self._qobj['circuits']:
+            if (('compiled_circuit_qasm' not in circuit) or
+                    (circuit['compiled_circuit_qasm'] is None)):
+                compiled_circuit = openquantumcompiler.compile(
+                    circuit['circuit'].qasm())
+                circuit['compiled_circuit_qasm'] = compiled_circuit.qasm(qeflag=True)
+            if isinstance(circuit['compiled_circuit_qasm'], bytes):
+                api_jobs.append({'qasm': circuit['compiled_circuit_qasm'].decode()})
+            else:
+                api_jobs.append({'qasm': circuit['compiled_circuit_qasm']})
+
+        seed0 = self._qobj['circuits'][0]['config']['seed']
+        output = self._api.run_job(api_jobs, self._qobj['config']['backend'],
+                             shots=self._qobj['config']['shots'],
+                             max_credits=self._qobj['config']['max_credits'],
+                             seed=seed0)
+        if 'error' in output:
+            raise ResultError(output['error'])
+
+        job_result = _wait_for_job(output['id'], self._api, wait=wait,
+                                   timeout=timeout)
+        job_result['name'] = self._qobj['id']
+        job_result['backend'] = self._qobj['config']['backend']
+        this_result = Result(job_result, self._qobj)
+        return this_result
+
+    @classmethod
+    def set_api(cls, api):
+        """Associate API with class"""
+        cls._api = api
```

Comment:
Would it be possible to define `_api` explicitly as a class variable too - adding `_api = None` in the definition of the class (after line ~11)? It would facilitate keeping track of what attributes are present on the class (in the same spirit as the convention of not defining instance attributes outside `__init__` noted on a comment below).

Reply:
Done.